src/gtc/dace/transformations.py

```diff
@@ -0,0 +1,112 @@
+import dace
+from dace.transformation.dataflow import TrivialMapElimination
+from dace.transformation.helpers import nest_state_subgraph
+from dace.transformation.interstate import InlineTransients
+
+
+def eliminate_trivial_maps(sdfg: dace.SDFG):
+    """Remove maps and map ranges where the iteration is over a single index."""
+    applied = True
+    while applied:
+        applied = False
+        for map_entry, state in sdfg.all_nodes_recursive():
+            if isinstance(map_entry, dace.nodes.MapEntry):
+                if map_entry.map.schedule in {
+                    dace.ScheduleType.Sequential,
+                    dace.ScheduleType.CPU_Multicore,
+                }:
+                    # exclude maps with empty edges as workaround for a bug in TrivialMapElimination
+                    if any(
+                        edge.data.is_empty()
+                        for edge in state.in_edges(map_entry)
+                        + state.out_edges(state.exit_node(map_entry))
+                    ):
```

Comment:
What does the + mean in this context? Is it adding a generator expression to another? :O

Reply:
lists can be joined with the + operator, we do this everywhere :)
I would like to avoid making this longer, the do something for inputs... and outputs... is everywhere in the dace backend and I'd like to keep the overhead that causes minimal.

Reply:
Ah ok, but it's not a list in this case, rather `edge.data.is_empty() for edge in state.in_edges(map_entry)` is a Generator expression. I wasn't aware that that `+` was overloaded for them the same way it is for Lists. It must be that there is something like an `itertools.chain` happening behind the scenes with a closure. Interesting...

Reply:
Ah, I see what you are saying. I don't think the `+` is overloaded like that, but this example has different associtaivity though: The  `+` operates on the operands `state.in_edges(map_entry)` and `state.out_edges(state.exit_node(map_entry))`. I put parantheses now to make it clearer to read (and `black` lets me keep them :sweat_smile: )